### PR TITLE
Issue 93 fix

### PR DESCRIPTION
**Pull request by amanwriter**
Registry will now only respond to pings from registry_client if its services are registered. 

This will ensure that the registry_client of an active but deregistered service will eventually re-register.

### CODE REVIEW

File: vyked/registry.py [hunk 1]
```diff
@@ -216,7 +216,17 @@ def receive(self, packet: dict, protocol, transport):
         elif request_type == 'pong':
             self._ping(packet)
         elif request_type == 'ping':
-            self._pong(packet, protocol)
+            if 'payload' in packet:
```
Comment (kashifrazzaqui) [inline vyked/registry.py]: What is this logic trying to do? How about a comment explaining the motive? Better still move this to a private method that has a docstring indicating the purpose of this.

Comment (amanwriter) [inline vyked/registry.py]: Is this okay?

```
def _handle_ping(self, packet, protocol):
    """ Responds to pings from registry_client only if the node_ids present in the ping payload are registered

    :param packet: The 'ping' packet received
    :param protocol: The protocol on which the pong should be sent
    """
    if 'payload' in packet:
        to_send=True
        node_ids = list(packet['payload'].values())
        for node_id in node_ids:
            if self._repository.get_node(node_id) is None:
                to_send = False
                break
        if to_send:
            self._pong(packet, protocol)
    else:
        self._pong(packet, protocol)
```

Comment (kashifrazzaqui) [inline vyked/registry.py]: Yup much more readable. Also _to_send_ could be renamed to _is_valid_node_

Comment (amanwriter) [inline vyked/registry.py]: the pong is sent to the registry_client not the node

Comment (kashifrazzaqui) [inline vyked/registry.py]: Yes, but you are checking if a node_id is valid or not before responding which is what the flag 'to_send' is for. A better name might be is_valid_node for the flag.

**Pull request merged**